LoopKitUI/ViewModels/CorrectionRangeScheduleEditorViewModel.swift

```diff
@@ -20,10 +20,17 @@ struct CorrectionRangeScheduleEditorViewModel {
 
     var saveGlucoseTargetRangeSchedule: (_ glucoseTargetRangeSchedule: GlucoseRangeSchedule) -> Void
 
-    init(therapySettingsViewModel: TherapySettingsViewModel,
-         didSave: (() -> Void)? = nil)
-    {
-        self.glucoseTargetRangeSchedule = therapySettingsViewModel.glucoseTargetRangeSchedule
+    init(
+        mode: SettingsPresentationMode,
+        therapySettingsViewModel: TherapySettingsViewModel,
+        didSave: (() -> Void)? = nil
+    ) {
+        if mode == .acceptanceFlow, let unsafeSchedule = therapySettingsViewModel.glucoseTargetRangeSchedule {
+            self.glucoseTargetRangeSchedule = unsafeSchedule.safeSchedule(with: therapySettingsViewModel.suspendThreshold?.value, unit: unsafeSchedule.unit)
```

Comment:
`therapySettingsViewModel.suspendThreshold` has its own, separate, units from `unsafeSchedule.unit` here. I'd recommend making `safeSchedule` take an HKQuantity, and you can then pass in `therapySettingsViewModel.suspendThreshold.quantity` to `safeSchedule`.  `safeSchedule`, since it is a method on `GlucoseRangeSchedule`, has access to `unit` locally, and doesn't need to have units passed in.

Reply:
I just implemented those changes!